### PR TITLE
removed jquery dependency

### PR DESCRIPTION
**Pull request by murb**
Previously @mdartic created a pull request to remove jQuery. Partly based on his request, cleaned and completed a new pull request that removes jQuery from the main libraries and most of the examples. 

Examples 1, 2, 3, 4, 6, 7, 8, 9, 10, 13, 14, 15 and 16 have been tested and found to be working.

Example 5 doesn't work (but is broken at http://apps.socib.es/ as well)
Example 11 doesn't work (but is broken at http://apps.socib.es/ as well)
Example 12 doesn't work (but is broken at http://apps.socib.es/ as well)

Examples 3 and 10 still use jQuery for other tasks.

### CODE REVIEW

**Comment by murb** [comment]
The related issue is https://github.com/socib/Leaflet.TimeDimension/issues/86 ; this pull request replaces https://github.com/socib/Leaflet.TimeDimension/pull/89 .

**Comment by bielfrontera** [comment]
Hi @murb,
many many thanks for your work. And @mdartic for his previous work!
As there are a lot of changes, let me check them and also check the broken examples. I expect I can merge soon!!

File: examples/example1.html [hunk 1]
```diff
@@ -51,15 +50,15 @@
...
-                        $('pre code').each(function(i, block) {
-                            hljs.highlightBlock(block);
-                        });
-                    });
-            });
+           var oReq = new XMLHttpRequest();
+           oReq.addEventListener("load", (function(xhr) {
+             var code = xhr.currentTarget.response;
+             var codeBlock = document.getElementById("code");
+             codeBlock.innerHTML = code;
```
Comment (bielfrontera) [inline examples/example1.html]: I think we should use `innerText`. There are some HTML within the javascript code, and innerText will prevent to render it as HTML.

File: examples/js/example12.js [hunk 2]
```diff
@@ -16,7 +16,6 @@ L.TimeDimension.Layer.SODAHeatMap = L.TimeDimension.Layer.extend({
             lngField: 'lng',
             valueField: 'count'
         };
-        heatmapCfg = $.extend({}, heatmapCfg, options.heatmatOptions || {});
```
Comment (bielfrontera) [inline examples/js/example12.js]: I will add a new function inside L.TimeDimension.Util to merge the default configuration with the options (this function: https://gomakethings.com/vanilla-javascript-version-of-jquery-extend/)

Btw, I have this example working after replacing `$.getJSON` by `XMLHttpRequest`

File: examples/example1.html (hunk 1 above)
Comment (murb) [reply examples/example1.html]: changed it to `textContent` instead: `innerText` strips whitespace, which is no problem until the first comment :)

File: examples/js/example12.js (hunk 2 above)
Comment (murb) [reply examples/js/example12.js]: makes sense, didn't bother replacing it here, as it was just an example and overriding configuration wasn't used as far as I could see.

Comment (r1m) [reply examples/js/example12.js]: [Object.assign](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) ? It is supported on all major browser and can be polyfill if needed

Comment (murb) [reply examples/js/example12.js]: btw. it is not yet necessary to implement either a polyfill and/or use Object assign. This example is the only place where I removed an `$.extend` bluntly, I replaced it with a simple `for`-loop here (see also https://stackoverflow.com/questions/171251/how-can-i-merge-properties-of-two-javascript-objects-dynamically#171256 for some discussion): https://github.com/socib/Leaflet.TimeDimension/pull/119/commits/ff96af7393b1f5c7c01c877f12f3b5be98e84a85#diff-1a71af7bca9eb8c6869b9fbf22fc0a1bL441 ...

**Comment by murb** [comment]
@bielfrontera: if there's anything I can do to help you, let me know :)

**Comment by bielfrontera** [comment]
Hi @murb, nice work!
I'm changing some examples and I will probably push the new release tomorrow!

**Pull request merged**